.github/workflows/testing_and_building_repo.yml

```diff
@@ -0,0 +1,110 @@
+name: testing_and_building_repo
+on: [push, pull_request]
+jobs:
+  build: 
+    strategy:
+      matrix:
+        os: ['ubuntu-18.04']
+
+    runs-on: ${{matrix.os}}
+    defaults:
+      run:
+        shell: bash -l -e -o pipefail {0}
+    env:
+      WSI_CONDA_CHANNEL: "https://dnap.cog.sanger.ac.uk/npg/conda/devel/generic"
+
+    name: Distribution Perl on ${{ matrix.os }} 
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: '0' 
+    
+    # Caching conda
+    - name: Cache conda
+      id: minicondaCache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/conda/pkgs
+          ~/conda/envs
+        key: ${{ matrix.os}}-build-miniconda
+    
+   # Caching cpanm external modules
+    - name: Cache cpanm external modules
+      id: cpanmCache
+      uses: actions/cache@v2
+      with:
+        path: ~/perl5ext  
+        key: ${{ matrix.os}}-build-cpanm-external
+
+    - name: Initialize Miniconda
+      run: |
+          echo 'source $CONDA/etc/profile.d/conda.sh' >> "$HOME/.bash_profile"
+
+    - name: install libgd-dev and uuid-dev 
+      run:
+          sudo apt-get install -y libgd-dev  uuid-dev 
+   
+    - name: install cpanm
+      run: |
+          wget -qO - https://cpanmin.us | /usr/bin/perl - --sudo App::cpanminus 
+
+    - name: run Before_install.sh
+      run: |
+          ${GITHUB_WORKSPACE}/scripts/before_install.sh
+
+    - name: install conda client, baton,samtools, set up conda environment
+      run: |
+          conda config --prepend pkgs_dirs ~/conda/pkgs
+          conda config --prepend envs_dirs ~/conda/envs
+
+          conda config --set auto_update_conda False
+          conda config --prepend channels "$WSI_CONDA_CHANNEL"
+          conda config --append channels conda-forge
+          conda info
+
+          conda create -y -n "$CONDA_TEST_ENV" 
+          conda install --name "$CONDA_TEST_ENV" biobambam2
```

Comment:
Please be consistent with previous line i.e. stick with `--name` or `-n`, but don't mix.